Client/src/components/navbar.tsx

```diff
@@ -0,0 +1,186 @@
+import '../css/navbar.css';
+import { useState, useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useAuth0 } from '@auth0/auth0-react';
+import MediaQuery from 'react-responsive';
+import apiService from '../services/apiService';
+import { useAppDispatch, useAppSelector } from '../features/hooks';
+import { refreshData } from '../features/user_slice';
+
+function Navbar() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [toggle, setToggle] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const { loginWithRedirect, logout, user, isLoading, isAuthenticated } =
+    useAuth0();
+
+  const toggleNavbar = () => {
+    setToggle(!toggle);
+  };
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    (async function evalAuth() {
+      if (isAuthenticated === true) {
+        const { user, message } = await apiService.profile();
+        console.log(user)
+        if (user) dispatch(refreshData(user));
```

Comment:
This user is undefined if the call to the backend goes wrong, so then it displays the error messgae

Reply:
Wrapped it in a try catch